source/app/scorearea.cpp

```diff
@@ -42,15 +42,19 @@ ScoreArea::ScoreArea(QWidget *parent)
     : QGraphicsView(parent),
       myScoreInfoBlock(nullptr),
       myCaretPainter(nullptr),
+      myScorePalette(parent->palette()),
```

Comment:
This is partially right: caching caused a problem for the background colour. However, the score itself is not redrawn using the new palette once the palette is changed. I fixed this locally: when the event "PaletteChange" happens, I will render the document, so it uses the new system palette automatically.

Reply:
Thanks for the details, that approach sounds good!

Reply:
For the OSX staff colors: the problem seems to be that every theme has a unique palette that does not always fit our criteria for score colors, at least partially. I came into this conclusion because the colors on ubuntu seem to be good, close to the image you have mentioned. 
I came up with a solution: I calculate the color of the staff as a weighted average between the background color and the text color, this results in the "faded" color of the staff lines. I currently do not have access to OSX right now, but at least in Ubuntu dark themes the result seems to be good for me.

Reply:
Yeah, I think the new behaviour looks better on OSX now!